pwiz_tools/Skyline/Model/GlobalizedObject.cs

```diff
@@ -139,7 +141,13 @@ public override bool Equals(object obj)
             var otherProps = other.GetPropertiesForComparison()
                 .ToDictionary(prop => prop.Name, prop => prop.GetValue(other));
 
-            return ArrayUtil.EqualsDeep(thisProps, otherProps);
+            var joinedValues = (from t in thisProps
+                join o in otherProps on t.Key equals o.Key
+                select new { t = t.Value, o = o.Value }).ToList();
+            if (joinedValues.Count != thisProps.Count)
+                return false;
+            var res = joinedValues.Any(tuple => !tuple.t.Equals(tuple.o));
+            return !res;
```

Comment:
We usually do not override "Equals" and "GetHashCode" for objects that are mutable.
That is, a mutable object is considered equal to something else if and only if they are the same object.
Weird bugs happen if you use a mutable object as the key in a hash map and the object gets changed later.

(It looks like this "Equals" method is only used for testing. I would recommend writing a different method "GetPropertyValues" which the property names and values as an ImmutableList sorted by name, and you can use ImmutableList equals method.

Reply:
There is a very little chance that these objects will be used as dictionary keys. Property values can be objects themselves, so the ImmutableList approach doesn't fit very well. Should I just use Serialize method and compare JSON strings?